src/VirtoCommerce.Platform.Core/Common/AsyncLock.cs

```diff
@@ -56,41 +56,54 @@ public void Release()
     // http://blogs.msdn.com/b/pfxteam/archive/2012/02/12/10266988.aspx
```

Comment:
This reference is expired. But we can use https://devblogs.microsoft.com/pfxteam/building-async-coordination-primitives-part-6-asynclock/. Does it sense?